src/redisraft.c

```diff
@@ -622,6 +628,21 @@ static bool handleInterceptedCommands(RedisRaftCtx *rr,
     return false;
 }
 
+/* TODO: used to only add entry for clients that have actually watched,
+ *       needed to avoid testing inconsistencies for now
+ */
+static void handleWatch(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
```

Comment:
Do we have to define `watch` as a client state? We already keep the information in watched sessions in `client_session_dict`, and we should use this dict in order to call `UNWATCH`, even in disconnection - otherwise, Redis will not update its watched_keys dict.

Reply:
this is to fix the testing issue of having possible inconsistent number of log entries in our tests.  Its not "needed", as we could always send the log entry and it wouldn't do harm to a real system, it just breaks tests that expect the number of entries to always be the same.

We can't rely on the client_session_dict because it can disconnect after append but before apply and then a client_session will be created and leak.  With that said, its possible we could be "ok" with said leak (will be cleaned up at election time, so not a permanent leak), and then we could just rely on the client_session_dict without this new entry.